pkg/lagoon/ssh/main.go

```diff
@@ -1,14 +1,40 @@
+// Package ssh implements an SSH client for Lagoon.
 package ssh
 
 import (
 	"bytes"
 	"fmt"
 	"os"
+	"strings"
 
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/term"
 )
 
+// LogStream connects to host:port using the given config, and executes the
+// argv command. It does not request a PTY, and instead just streams the
+// response to the attached terminal. argv should contain a logs=... argument.
+func LogStream(config *ssh.ClientConfig, host, port string, argv []string) error {
+	// https://stackoverflow.com/a/37088088
+	client, err := ssh.Dial("tcp", host+":"+port, config)
+	if err != nil {
+		return fmt.Errorf("couldn't dial SSH: %v", err)
```

Comment:
I don't think overriding the error returned by Dial is a good option, but you could add a note in here about how maybe the service doesn't support the `logs` command.

Reply:
Yeah, I wasn't going to override the dial error just extend it with something like this. but then some dial errors could be completely unrelated to the log endpoint and so could confuse the error.
```
fmt.Errorf("couldn't dial SSH: %v\nthis error could mean that the requested endpoint does not currently support log streaming", err)
````
If you have a better suggestion?

Reply:
Longer term, I'm planning to implement a version in the banner to address https://github.com/uselagoon/lagoon/issues/3693, so the Lagoon CLI could use a [BannerCallback](https://pkg.go.dev/golang.org/x/crypto/ssh#BannerCallback) to check the version?

Reply:
Sure, but the banner might have to also describe if the logs endpoint is enabled or not too?

Reply:
Is this enough to get the point across?

```
"couldn't dial SSH (maybe this service doesn't support logs?): %v"
```

> Sure, but the banner might have to also describe if the logs endpoint is enabled or not too?

Ah good point. That idea probably won't work then.

Reply:
yeh that error message works for me for now. clear enough to get the point